tests/chainerx_tests/unit_tests/routines_tests/test_indexing.py

```diff
@@ -214,31 +259,38 @@ def forward_xp(self, inputs, xp):
     ((2, 3), (2, 3), (1, 3)),
 ])
 @chainer.testing.parameterize_pytest(
-    'dtypes', chainerx.testing.all_dtypes)
+    'in_dtypes,out_dtype', _in_out_dtypes_where
+)
 @chainer.testing.parameterize_pytest(
     'condition_dtypes', chainerx.testing.integral_dtypes)
 class TestWhere(op_utils.NumpyOpTest):
 
+    check_numpy_strides_compliance = False
+
     def setup(self):
-        if numpy.dtype(self.dtypes).kind != 'f':
+        (x_dtype, y_dtype) = self.in_dtypes
+        if numpy.dtype(x_dtype).kind != 'f' or \
+           numpy.dtype(y_dtype).kind != 'f':
             self.skip_backward_test = True
             self.skip_double_backward_test = True
 
-        if self.dtypes == 'float16':
+        if x_dtype == 'float16' or y_dtype == 'float16':
             self.check_backward_options.update({'rtol': 1e-3, 'atol': 1e-3})
 
     def generate_inputs(self):
-        x = numpy.random.uniform(-1, 1, self.x_shape).astype(self.dtypes)
-        y = numpy.random.uniform(-1, 1, self.y_shape).astype(self.dtypes)
+        (x_dtype, y_dtype) = self.in_dtypes
+        x = numpy.random.uniform(-1, 1, self.x_shape).astype(x_dtype)
+        y = numpy.random.uniform(-1, 1, self.y_shape).astype(y_dtype)
         condition = \
             numpy.random.uniform(-1, 1,
                                  self.cond_shape).astype(self.condition_dtypes)
         return (x, y, condition)
 
     def forward_xp(self, inputs, xp):
         x, y, condition = inputs
```

Comment:
The given condition might be better to be of bool dtype. How about operating >0.5 in `generate_inputs`?

Reply:
Moved it to `generate_inputs` and cast `condition > 0.5` to the `condition_dtype`. 
Earlier input `condition` to `xp.where` was always bool as you mentioned. Thanks.